tests/shell/check_meta.sh

```diff
@@ -0,0 +1,39 @@
+@INCLUDE_COMMON@
+
+echo
+echo ELEKTRA PLUGIN METATADA CONSISTENCY CHECK
+echo
+
+PLUGINS_DIR="@CMAKE_SOURCE_DIR@/src/plugins"
+META_FILE="@CMAKE_SOURCE_DIR@/doc/METADATA.ini"
+
+for PLUGIN in `ls "$PLUGINS_DIR"`;
+do
+
+    if [ ! -d "${PLUGINS_DIR}/${PLUGIN}" ];
+    then
+	continue
+    fi
+    
+    README="${PLUGINS_DIR}/${PLUGIN}/README.md"
+    PLUGIN_META=$(grep -Eo "infos/metadata(.*)" "$README" |cut -d '=' -f2)
+    for META in $PLUGIN_META;
+    do
+	grep -Eq "^\\[${META}\\]$" "$META_FILE"
+        succeed_if "Metadata $META of plugin $PLUGIN not present in METADATA.ini"
+    done
+    USED_BY=$(awk "/usedby\\/plugin= ([^\\n]*)${PLUGIN}([^\\n]*)/" RS= "$META_FILE")
```

Comment:
Looks like `gawk` uses `\b` as shortcut for [backspace](https://www.gnu.org/software/gawk/manual/html_node/GNU-Regexp-Operators.html). In theory `\y` should work, but as `\y` is a `gawk` specific regex operator, using it will break the command if someone uses another version of `awk`.

I assume the pattern `( |$)` works. We need to escape the `$` sign though. I tried the pattern `"/^usedby\\/plugin= .*$PLUGIN( |\$)/"`, which seems to work:

```sh
set PLUGIN calculate
awk "/^usedby\\/plugin= .*$PLUGIN( |\$)/" doc/METADATA.ini
# usedby/plugin= calculate

set PLUGIN conditionals
awk "/^usedby\\/plugin= .*$PLUGIN( |\$)/" doc/METADATA.ini
# usedby/plugin= conditionals

set PLUGIN c
awk "/^usedby\\/plugin= .*$PLUGIN( |\$)/" doc/METADATA.ini

set PLUGIN validation
awk "/^usedby\\/plugin= .*$PLUGIN( |\$)/" doc/METADATA.ini
# usedby/plugin= validation
# usedby/plugin= validation
# usedby/plugin= validation
# usedby/plugin= validation
# usedby/plugin= validation

set PLUGIN augeas
awk "/^usedby\\/plugin= .*$PLUGIN( |\$)/" doc/METADATA.ini
# usedby/plugin= hosts augeas keytometa

set PLUGIN keytometa
awk "/^usedby\\/plugin= .*$PLUGIN( |\$)/" doc/METADATA.ini
# usedby/plugin= hosts augeas keytometa
```
.